src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Initializers/WebJobsTelemetryInitializer.cs

```diff
@@ -49,47 +49,61 @@ public void Initialize(ITelemetry telemetry)
+                // Apply our special scope properties
+                IDictionary<string, object> scopeProps = DictionaryLoggerScope.GetMergedStateDictionaryOrNull();
 
-                // Apply Category and LogLevel to all telemetry
-                string category = scopeProps.GetValueOrDefault<string>(LogConstants.CategoryNameKey);
-                if (category != null)
+                string invocationId = scopeProps?.GetValueOrDefault<string>(ScopeKeys.FunctionInvocationId);
+                if (invocationId != null)
                 {
-                    telemetryProps[LogConstants.CategoryNameKey] = category;
+                    telemetryContext.Properties[LogConstants.InvocationIdKey] = invocationId;
                 }
 
-                LogLevel? logLevel = scopeProps.GetValueOrDefault<LogLevel?>(LogConstants.LogLevelKey);
-                if (logLevel != null)
+                // this could be telemetry tracked in scope of function call - then we should apply the logger scope
+                // or RequestTelemetry tracked by the WebJobs SDK or AppInsight SDK - then we should apply Activity.Tags
+                if (scopeProps != null && scopeProps.Count > 0)
                 {
-                    telemetryProps[LogConstants.LogLevelKey] = logLevel.Value.ToString();
-                }
+                    telemetryContext.Operation.Name = scopeProps.GetValueOrDefault<string>(ScopeKeys.FunctionName);
 
-                int? eventId = scopeProps.GetValueOrDefault<int?>(LogConstants.EventIdKey);
-                if (eventId != null && eventId.HasValue && eventId.Value != 0)
-                {
-                    telemetryProps[LogConstants.EventIdKey] = eventId.Value.ToString();
-                }
+                    // Apply Category, LogLevel event details to all telemetry
+                    if (!telemetryContext.Properties.ContainsKey(LogConstants.CategoryNameKey))
+                    {
+                        string category = scopeProps.GetValueOrDefault<string>(LogConstants.CategoryNameKey);
+                        if (category != null)
+                        {
+                            telemetryContext.Properties[LogConstants.CategoryNameKey] = category;
+                        }
+                    }
 
-                string eventName = scopeProps.GetValueOrDefault<string>(LogConstants.EventNameKey);
-                if (eventName != null)
-                {
-                    telemetryProps[LogConstants.EventNameKey] = eventName;
+                    if (!telemetryContext.Properties.ContainsKey(LogConstants.LogLevelKey))
+                    {
+                        LogLevel? logLevel = scopeProps.GetValueOrDefault<LogLevel?>(LogConstants.LogLevelKey);
+                        if (logLevel != null)
+                        {
+                            telemetryContext.Properties[LogConstants.LogLevelKey] = logLevel.Value.ToString();
```

Comment:
Can use the switched .ToString method here to avoid an allocation

Reply:
Done.